### PR TITLE
[backend/frontend] Retention rules for files and workbench (#7482)

### LINKED ISSUE #7482: retention for FILES MVP
## Use case

Ensure that we have a short term solution for this https://github.com/OpenCTI-Platform/opencti/issues/3016

### PR DESCRIPTION
**Pull request by Archidoit**
### Proposed changes
Possibility to create retention rules on global files and workbenches : 
- UI improvement
- possibility to create retention rules on files and workbenches
- retention manager adaptation to handle retention on files and workbenches

Under feature flag: all the new frontend implementation

### Related issues
https://github.com/OpenCTI-Platform/opencti/issues/7482

## Related docs PR
https://github.com/OpenCTI-Platform/docs/pull/200

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/settings/retention/RetentionEdition.jsx [hunk 1]
```diff
@@ -173,44 +172,48 @@ const RetentionEditionContainer = (props) => {
...
-                  'fromTypes',
-                  'toTypes',
-                ]}
+            {retentionRule.scope === 'knowledge' && <>
+              <Box sx={{
+                paddingTop: 4,
+                display: 'flex',
+                gap: 1,
+              }}
+              >
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/settings/retention/RetentionEdition.jsx]: strange indent 🤔

File: opencti-platform/opencti-graphql/src/config/conf.js [hunk 2]
```diff
@@ -419,7 +419,6 @@ export const GRAPHQL_ARMOR_ENABLED = booleanConf('app:graphql:armor_enabled', tr
 export const ENABLED_TRACING = booleanConf('app:telemetry:tracing:enabled', false);
 export const ENABLED_METRICS = booleanConf('app:telemetry:metrics:enabled', false);
 export const ENABLED_EVENT_LOOP_MONITORING = booleanConf('app:event_loop_logs:enabled', false);
-export const ENABLED_RETENTION_MANAGER = booleanConf('retention_manager:enabled', true);
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/config/conf.js:422]: why this change ? var never used ?

File: opencti-platform/opencti-graphql/src/database/file-storage-helper.ts [hunk 3]
```diff
@@ -133,6 +133,7 @@ export const deleteAllBucketContent = async (context: AuthContext, user: AuthUse
   }
 
   // Once all files are deleted, then bucket can be removed.
+  // -- commented because not working --
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/database/file-storage-helper.ts:136]: do you have more details ? an issue link ?

File: opencti-platform/opencti-graphql/src/config/conf.js (hunk 2 above)
Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/config/conf.js:422]: yes unused since I migrated the retentionManager under the new manager format

File: opencti-platform/opencti-front/src/private/components/settings/retention/RetentionEdition.jsx (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/settings/retention/RetentionEdition.jsx]: fixed

File: opencti-platform/opencti-graphql/src/database/file-storage-helper.ts (hunk 3 above)
Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/database/file-storage-helper.ts:136]: it's juste a precision regarding the commented line below (seen with @aHenryJard )

**Review by labo-flg — CHANGES_REQUESTED** [review]
Rename some functions that are unclear
* deleteElements > deleteElement (singular)
* elementsToDelete > getElementsToDelete (action verb is way more clear for all functions)

Add some tests if possible to the function deleteElement too. Rest of the manager (execute, handler) is difficult to test.

**Review by labo-flg — APPROVED** [review]
